packages/core/lib/patchers/aws3_p.ts

```diff
@@ -110,15 +110,21 @@ const getXRayMiddleware = (config: RegionResolvedConfig, manualSegment?: Segment
     return next(args);
   }
 
-  const subsegment: Subsegment = segment.addNewSubsegment(service);
+  let subsegment: Subsegment;
+  if (segment.notTraced == false || !segment.subsegments[segment.subsegments.length - 1].notTraced) {
```

Comment:
Would this also work?
if (!segment.subsegments[segment.subsegments.length - 1].notTraced)

Reply:
Yes it would! Will incorporate that change

Reply:
The reason this is being kept in is to account for subsegments that only have a parent segment (no parent subsegments). After some debugging, I discovered that this line 

`const segment = contextUtils.isAutomaticMode() ? contextUtils.resolveSegment() : manualSegment;`

always returns the **currently active segment**. If there are no active subsegments and the parent segment is unsampled, this check will not behave correctly without the `segment.notTraced == false`

Reply:
This is more of a design decision: check the parent sampling decision here for unsampled segments, or only check the subsegment and rely on the emitter not sending the entire segment if it is unsampled down the line. If the latter was used in the other SDKs, I am happy to incorporate the same design here as well.